frontend/model/captureLogs.js

```diff
@@ -0,0 +1,176 @@
+import sbp from '~/shared/sbp.js'
+import { CAPTURED_LOGS, SET_APP_LOGS_FILTER } from '~/frontend/utils/events.js'
+
+/*
+  giConsole/count - total logs stored
+  giConsole/lastEntry - latest log hash.
+  giConsole/markers - an array of markers at every 1000th entry.
+*/
+const ENTRIES_LIMIT = 4000
+const ENTRIES_MARKER = 1000
+
+const getMarkers = () => JSON.parse(localStorage.getItem('giConsole/markers')) || []
+
+// these are initialized at captureLogsStart()
+let isCapturing = false
+let isConsoleOveridden = false
+let lastEntry = null
+let entriesCount = null
+let appLogsFilter = []
+
+function captureLog (type, ...msg) {
+  const logEntry = `${Date.now()}_${Math.floor(Math.random() * 100000)}` // uuid
+
+  // detect when is an Error and capture it properly
+  // ex: uncaught Vue errors or custom try/catch errors.
+  msg = msg.map((m) => m instanceof Error ? (JSON.stringify(m.stack) || m) : m)
```

Comment:
On second thought, I think it should be safe to do `m.stack` instead of `JSON.stringify(m.stack)`, since we call `JSON.stringify` below before storing the `msg`

Reply:
I thought the same but no... `JSON.stringify()` will ignore objects that are `instanceof Error` and output only an empty object `{}`. That's why I did the Error "detection" map before.

Reply:
Ah true, but that's why there's a `m instanceof Error` here, and `m.stack` isn't an Error object?

Reply:
`m` is an "error object", but it can still be accessed through 2 "special" keys as a normal object: `stack` and `message`. We access `m.stack` to have a more complete log. And `m.stack` is now a regular object that can be Stringified.

... You are right, JS is crazy sometimes 😄

Reply:
I've given this more thought, and I'm thinking that for `captureLog` we want `m.message` instead of `m.stack`, because it will probably be more helpful. Can you give it a test and reply here to show the difference?

Reply:
No, it's not more useful. m.message only contains the messag.e m.stack contains the message and where it happened. Here's the difference between both: 
![image](https://user-images.githubusercontent.com/14869087/77537788-94deca80-6e96-11ea-80df-8ed3f9b034ba.png) Knowing the error happened at "UserSettingsModal.js" is super useful in my opinion.

Reply:
Sorry, I meant a screenshot of what it looks like when it's stingified. That makes all the difference.

Reply:
copying from slack:

> cause i'm curious if the console output has some sort of special handling for m.stack, so we need to make sure that JSON.stringify(m.stack) still contains the original error message
> if it does, then yeah it's OK to keep it as m.stack

If you find out that `m.stack` is simply a string and that `JSON.stringify(m.stack)` still contains the error message, then we can stick with `m.stack`, in which case just change the code above from `JSON.stringify(m.stack)` to `m.stack`. I'm just trying to make sure we're not being fooled by fancy console output stuff that browsers have, and that the data we need is in fact being preserved.

Reply:
Yes, m.stack is simply a string and it contains the error message.

Reply:
OK great! Then the above can be changed from `(JSON.stringify(m.stack) || m)` => `(m.stack || m)`

Reply:
Oh! Event better:

From: `(JSON.stringify(m.stack) || m)`

To: `(m.stack || m.message)`

Because I'm pretty sure one of those will be defined across all environments.